### DISCUSSION TITLE
pathToFileURL is not a function with Web Worker in Webpack 5

### DISCUSSION THREAD

**Discussion by pedroteixeira** [webpack/webpack#17081] [Q&A] [answered]
Hi,

Attempting to follow instructions from the documentation site to have a Web Worker with Webpack 5 (in a .ts file, using babel):

```
const worker = new Worker(new URL('./file.worker.js', import.meta.url))
```

latest version 5.81.0 yields an exception `webpack_require__(...).pathToFileURL is not a function` 

not sure if related to Webpack 5 not exporting built-in node module `url`  anymore, I didn't understand what's the way to make this work.

I see some browser polyfills for `url` module but which don't have the function `pathToFileURL`

It seems Webpack didn't convert this code correctly:

If using `resolve.fallback.url: false`:

```
const worker = new Worker(new URL('./file.worker.js', (__webpack_require__(/*! url */ "?06cf").pathToFileURL)(__filename).toString()));
```

else if the `url` module is available (from https://github.com/defunctzombie/node-url):

```
const worker = new Worker(new URL('./EventBuffer.worker.js', __webpack_require__(/*! url */ "../../node_modules/url/url.js").pathToFileURL(__filename).toString()));
```

Any suggestions, known hints?

Thanks,

**Comment by alexander-akait** [comment]
What is target?

**Reply by pedroteixeira** [reply]
I didn't explicitly set a target, assuming `target: 'web'` as default

```
    output: {
        filename: '[name].js',
        path: path.join(__dirname, '../public/assets/'),
        publicPath: '/public/assets/',
        chunkFilename: '[name].[chunkhash].js',
        globalObject: 'this'
    },
```

Using latest Babel libraries `7.21.4` - not sure if related.

**Reply by alexander-akait** [reply]
Can you provide full configuration, because it looks like you try to buidle application for node.js usage

**Reply by pedroteixeira** [reply]
I've added `target: web` but didn't change generated js. Any other hint appreciated. But I'll try to reduce config to something smaller to help troubleshooting, thanks for replying and about the possible root cause :crossed_fingers:

**Reply by pedroteixeira** [reply, marked as answer]
For some the project had babel using `babel-plugin-transform-import-meta` so it was transforming the code before Webpack - just need to remove it / or if needed for jest tests, just enable it in test env
